src/fromager/commands/find_updates.py

```diff
@@ -0,0 +1,255 @@
+import csv
+import json
+import logging
+import pathlib
+import sys
+from enum import Enum
+from typing import TextIO
+
+import click
+from packaging.requirements import Requirement
+from packaging.version import Version
+
+from fromager import constraints, context, overrides, resolver
+from fromager.commands.list_versions import DistributionType
+
+logger = logging.getLogger(__name__)
+
+
+class OutputFormat(Enum):
+    """Output format for find-updates command"""
+
+    REQUIREMENTS = "requirements"
+    JSON = "json"
+    CSV = "csv"
+
+
+@click.command()
+@click.option(
+    "--format",
+    "output_format",
+    type=click.Choice(OutputFormat, case_sensitive=False),
+    default=OutputFormat.REQUIREMENTS.value,
+    help="Output format (requirements: name==version, json: JSON array, csv: CSV with name,version columns)",
+)
+@click.option(
+    "--distribution-type",
+    type=click.Choice(DistributionType, case_sensitive=False),
+    default=DistributionType.DEFAULT.value,
+    help="Distribution type to include in version lookup (default: use package settings, sdist: source only, wheel: wheels only, both: include both sdists and wheels)",
+)
+@click.option(
+    "--sdist-server-url",
+    default=resolver.PYPI_SERVER_URL,
+    help="URL to the Python package index to use for version lookup",
+)
+@click.option(
+    "-o",
+    "--output",
+    type=click.Path(path_type=pathlib.Path),
+    help="Write output to file instead of stdout",
+)
+@click.argument(
+    "constraints_file", type=click.Path(exists=True, path_type=pathlib.Path)
+)
+@click.pass_obj
+def find_updates(
+    wkctx: context.WorkContext,
+    constraints_file: pathlib.Path,
+    output_format: str,
+    distribution_type: str,
+    sdist_server_url: str,
+    output: pathlib.Path | None,
+) -> None:
+    """Find available updates for packages with specific version constraints.
+
+    This command reads a constraints file and for each package that has a
+    supported constraint (==, <, <=), it lists newer versions available.
+    Constraints with lower bounds (>=) are ignored as they don't pin to specific versions.
+
+    The CONSTRAINTS_FILE should be a pip-style constraints file with entries like:
+    - "package_name==1.0.0" (will look for versions > 1.0.0)
+    - "package_name<2.0.0" (will look for versions >= 2.0.0)
+    - "package_name<=1.5.0" (will look for versions > 1.5.0)
+    - "package_name>=1.0.0" (ignored - not a specific constraint)
```

Comment:
I am a bit confused about the text here :sweat_smile: 
Shouldn't `package_name==1.0.0` means it will look for exactly that version? The text says it will look for > 1.0.0

Am I missing anything here?

Reply:
Good question. The goal of the command is to take an existing constraint and figure out if there are newer versions of packages that are being blocked because of that constraint. So the logic here is reversing the constraint, taking into account that we don't want to skip anything (we want a `<2.0.0` rule to look for versions starting with `>=2.0.0` for example, rather than `>2.0.0`).

Reply:
Ah, okay. Now I understand it better. Thank you!